contrib/cirrus/lib.sh

```diff
@@ -290,3 +290,33 @@ remove_packaged_podman_files() {
     # Be super extra sure and careful vs performant and completely safe
     sync && echo 3 > /proc/sys/vm/drop_caches || true
 }
+
+# Execute make localbenchmarks in $CIRRUS_WORKING_DIR/data
+# for preserving as a task artifact.
+localbenchmarks() {
+    local datadir
+    req_env_vars DISTRO_NV PODBIN_NAME PRIV_NAME TEST_ENVIRON TEST_FLAVOR
+    req_env_vars VM_IMAGE_NAME EC2_INST_TYPE
+
+    datadir=$CIRRUS_WORKING_DIR/data
+    mkdir -p $datadir
+
+    (
+      echo "# Metadata reflecting benchmark basis."
+      echo "# Intended for machine consumption."
+
+      printenv
+
+      echo"\
+INST_TYPE=$EC2_INST_TYPE  # one day may include other cloud's VM types.
+CPUTOTAL=$(grep -ce '^processor' /proc/cpuinfo)
+MEMTOTAL=$(awk -F: '$1 == "MemTotal" { print $2 }' </proc/meminfo | sed -e "s/^ *//")
+UNAME_RM=$(uname -r -m)
+"
+
+    ) | grep -Ev "$SECRET_ENV_RE" | sort > $datadir/benchmarks.env
```

Comment:
This is un-ideal. It will mingle the non-environment with real-environment, making it hard for a future reader. See above. Sorry for not having been clearer.